idaes/core/util/model_statistics.py

```diff
@@ -56,10 +56,16 @@ def number_total_blocks(block):
     Returns:
         Number of Block components in block (including block itself)
     """
-    b = 1  # Start at 1 to include main model
-    for o in block.component_data_objects(ctype=Block, active=None, descend_into=True):
-        b += 1
-    return b
+    # +1 to include main model
+    return (
+        sum(
+            1
+            for _ in block.component_data_objects(
+                ctype=Block, active=None, descend_into=True
+            )
+        )
+        + 1
+    )
```

Comment:
Would the following work instead of the `sum(1 for ...)`?

```suggestion
    objs = list(block.component_data_objects(ctype=Block, active=None, descend_into=True))
    # +1 to include main model
    return len(objs) + 1
```

Reply:
From what I read the sum is more efficient than casting to a list first, but it would work.